egs/iban/s5/local/nnet3/run_ivector_common.sh

```diff
@@ -0,0 +1,71 @@
+#!/bin/bash
+
+# this script is called from scripts like run_ms.sh; it does the common stages
+# of the build, such as feature extraction.
+# This is actually the same as local/online/run_nnet2_common.sh, except
+# for the directory names.
+
+. cmd.sh
+mfccdir=mfcc
+
+stage=1
+
+. cmd.sh
+. ./path.sh
+. ./utils/parse_options.sh
+
+
+if [ $stage -le 1 ]; then
+  for datadir in train dev; do
+    utils/copy_data_dir.sh data/$datadir data/${datadir}_hires
+    steps/make_mfcc.sh --nj 6 --mfcc-config conf/mfcc_hires.conf \
+      --cmd "$train_cmd" data/${datadir}_hires exp/make_hires/$datadir $mfccdir || exit 1;
+    steps/compute_cmvn_stats.sh data/${datadir}_hires exp/make_hires/$datadir $mfccdir || exit 1;
+  done
+fi
+
+mkdir -p exp/nnet3
+
+if [ $stage -le 2 ]; then
+  steps/align_fmllr.sh --nj 16 --cmd "$train_cmd" \
+    data/train data/lang exp/tri3b exp/nnet3/tri3b_ali
+fi
+
+if [ $stage -le 3 ]; then
+  # Train a small system just for its LDA+MLLT transform.  We use --num-iters 13
+  # because after we get the transform (12th iter is the last), any further
+  # training is pointless.
+  steps/train_lda_mllt.sh --cmd "$train_cmd" --num-iters 13 \
+    --realign-iters "" --splice-opts "--left-context=3 --right-context=3" \
+    5000 10000 data/train_hires data/lang \
+     exp/nnet3/tri3b_ali exp/nnet3/tri5b
+fi
+
+if [ $stage -le 4 ]; then
+  steps/online/nnet2/train_diag_ubm.sh --cmd "$train_cmd" --nj 16  --num-frames 200000 \
+     data/train_hires 256 exp/nnet3/tri5b exp/nnet3/diag_ubm
+fi
+
+if [ $stage -le 5 ]; then
+  # even though $nj is just 10, each job uses multiple processes and threads.
+ bash -x steps/online/nnet2/train_ivector_extractor.sh --cmd "$train_cmd" \
+    --nj 10 --num-processes 1 --num-threads 2  --ivector-dim 50\
```

Comment:
why the reduced dimension of ivectors ? I haven't seen an nnet2/nnet3 experiments which measures the impact of ivector dimension. If you want to reduce parameters in the model you could probably reduce the splicing config.